### PR TITLE
Added normalized laplacian

### LINKED ISSUE #283: Add normalized laplacian
Add definition of normalized laplacian as in "Learning with Hypergraphs: Clustering, Classification, and Embedding"

### PR DESCRIPTION
**Pull request by nwlandry**
* Added the normalized laplacian function. Fixes #283.
* Changed the default for `H.nodes.isolates()` to not ignore singletons.
* Fixed some of the documentation.

### CODE REVIEW

**Comment by nwlandry** [comment]
> > Changed the default for H.nodes.isolates() to not ignore singletons.
> 
> Why?

Because several references define an isolated node as degree 0, so I think this should be the default behavior. E.g., https://en.m.wikipedia.org/wiki/Vertex_(graph_theory)

**Comment by nwlandry** [comment]
> Thanks Nich. My only comment is that I wonder if the name "normalized Laplacian" is not too generic.

Something like "zhou_normalized_laplacian"? Did you have something in mind?

**Comment by leotrs** [comment]
> > > Changed the default for H.nodes.isolates() to not ignore singletons.
> > 
> > 
> > Why?
> 
> Because several references define an isolated node as degree 0, so I think this should be the default behavior. E.g., https://en.m.wikipedia.org/wiki/Vertex_(graph_theory)

The source you quote is for graph theory, in which case it's only relevant to our case if the graph allows self-loops.

I don't particularly care about which value is the default, but changing default values in general should have a bit of discussion or at least an explanation :sweat_smile:

**Review by leotrs — APPROVED** [review]

File: xgi/linalg/matrix.py [hunk 1]
```diff
@@ -370,6 +373,61 @@ def multiorder_laplacian(
...
+            "Every node must be a member of an edge to avoid divide by zero error!"
+        )
+
+    D = degree_matrix(H)
+    A, rowdict = clique_motif_matrix(H, sparse=sparse, index=True)
+
+    if sparse:
+        Dinvsqrt = csr_array(diags(np.power(D, -0.5)))
+        I = csr_array((H.num_nodes, H.num_nodes))
+        I.setdiag(1)
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/linalg/matrix.py:422]: The thing about this is that it makes a csr_matrix, not a csr_array :(

[the remark this answers is not in the document]
Comment (leotrs) [reply xgi/linalg/matrix.py:422]: Ohhh good catch. Yikes that's stupid. They'll probably change it later? Who knows.

File: xgi/linalg/matrix.py [hunk 2]
```diff
@@ -370,6 +373,61 @@ def multiorder_laplacian(
     return (L_multi, rowdict) if index else L_multi
 
 
+def normalized_laplacian(H, sparse=True, index=False):
+    """Compute the normalized Laplacian.
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/linalg/matrix.py:377]: I looked at the paper again and they call it the "hypergraph laplacian" so I settled on `normalized_hypergraph_laplacian`.

**Pull request merged**